rust_bindings/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(dead_code)]
+#![allow(non_upper_case_globals)]
+
+extern crate link_cplusplus;
+
+mod bindings {
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+}
+
+pub fn create_discriminant<const SIZE: usize>(seed: &[u8]) -> Option<[u8; SIZE]> {
+    // SAFETY: The length is guaranteed to match the actual length of the char pointer.
+    unsafe {
+        let mut result = [0; SIZE];
+        if !bindings::create_discriminant_wrapper(
+            seed.as_ptr(),
+            seed.len(),
+            SIZE * 8,
+            result.as_mut_ptr(),
+        ) {
+            return None;
+        }
+        Some(result)
+    }
+}
+
+pub fn verify_n_wesolowski(
+    discriminant: &[u8],
+    x_s: &[u8],
+    proof: &[u8],
+    num_iterations: u64,
+    recursion: u64,
+) -> bool {
+    // SAFETY: The lengths are guaranteed to match the actual lengths of the char pointers.
+    unsafe {
+        bindings::verify_n_wesolowski_wrapper(
+            discriminant.as_ptr(),
+            discriminant.len() * 8,
```

Comment:
it looks like the size is in bytes to begin with, and you're guaranteed to have a size divisible by 8. Wouldn't it be simpler to just pass the size in bytes rather than bits?